app/javascript/components/rooms/room/join/RoomJoin.jsx

```diff
@@ -100,15 +101,42 @@ export default function RoomJoin() {
     }
   }, [roomStatusAPI.isSuccess]);
 
+  // Plays a notify sound when the meeting starts if the user was in a waiting queue
+  const playNotifySound = () => {
+    const audio = new Audio(`data:audio/mpeg;base64,${NotifySound}`);
+    audio.play()
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  // Returns a random delay between 2 and 5 seconds, in increments of 250 ms
+  // The delay is to let the BBB server settle before attempting to join the meeting
+  // The randomness is to prevent multiple users from joining the meeting at the same time
+  const joinDelay = () => {
+    const min = 2000;
+    const max = 5000;
+    const step = 250;
+
+    // Calculate the number of possible steps within the given range
+    const numSteps = (max - min) / step;
+
+    // Generate a random integer from 0 to numSteps (inclusive)
+    const randomStep = Math.floor(Math.random() * (numSteps + 1));
+
+    // Calculate and return the random delay
+    return min + (randomStep * step);
+  };
+
   useEffect(() => {
     // Meeting started:
     //  When meeting starts this logic will be fired, indicating the event to waiting users (through a toast) for UX matter.
     //  Logging the event for debugging purposes and refetching the join logic with the user's given input (name & codes).
-    //  With a delay of 7s to give reasonable time for the meeting to fully start on the BBB server.
     if (hasStarted) {
       toast.success(t('toast.success.room.meeting_started'));
```

Comment:
Can we move this into the handleJoin as well?

Reply:
Yep